src/Model.jl

```diff
@@ -216,4 +226,37 @@ end
 spin_components(model::Model) = spin_components(model.spin_polarization)
 
 
-_is_well_conditioned(A; tol=1e5) = (cond(A) <= tol)
+# prevent broadcast
+import Base.Broadcast.broadcastable
+Base.Broadcast.broadcastable(model::Model) = Ref(model)
+
+
+#=
+There are two types of quantities, depending on how they transform under change of coordinates.
+
+Positions transform with the lattice: r_cart = lattice * r_red. We term them vectors.
+
+Linear forms on vectors (anything that appears in an expression f⋅r)
+transform with the inverse lattice transpose: f_cart = lattice' \ f_red. We term them covectors.
+Examples of covectors are forces.
+
+Reciprocal vectors are a special case: they are covectors, but conventionally have an
+additional factor of 2π in their definition, so they transform rather with 2π times the
+inverse lattice transpose: q_cart = 2π lattice' \ q_red = recip_lattice * q_red.
+=#
+vector_red_to_cart(model::Model, rred) = model.lattice * rred
+vector_cart_to_red(model::Model, rcart) = model.inv_lattice * rcart
+covector_red_to_cart(model::Model, fred) = model.inv_lattice' * fred
+covector_cart_to_red(model::Model, fcart) = model.lattice' * fcart
+recip_vector_red_to_cart(model::Model, qred) = model.recip_lattice * qred
+recip_vector_cart_to_red(model::Model, qcart) = model.inv_recip_lattice * qcart
+
+#=
+Transformations on vectors and covectors are matrices and comatrices. 
+Examples of matrices are the symmetries in real space (W)
+Examples of comatrices are the symmetries in reciprocal space (S)
+=#
+matrix_red_to_cart(model::Model, Ared) = model.lattice * Ared * model.inv_lattice
+matrix_cart_to_red(model::Model, Acart) = model.inv_lattice * Acart * model.lattice
+comatrix_red_to_cart(model::Model, Bred) = model.inv_lattice' * Bred * model.lattice'
+comatrix_cart_to_red(model::Model, Bcart) = model.lattice' * Bcart * model.inv_lattice'
```

Comment:
Maybe for consistency we should also have a `recip_matrix_cart_to_red` ? (Of course that would just call the `comatrix` function)

Reply:
recip_matrix is weird. This routine is used for transformations of (real and reciprocal) space, so when using that it's good to think of reciprocal space vectors as covectors, and so having the same routine helps